[REVIEW] Feature: Docs improvement

danielhanchen opened the pull request:
I'm a bit confused on how to merge to branch 0.6, so I'm trying again.

Sorry if I cause issues.
[I didn't change any code]

dantegd commented:
@danielhanchen this PR looks good from the standpoint of having no conflicts, thanks for resubmitting!!

@robocopnixon This PR improves documentation significantly, would you take a look?

robocopnixon commented:
The only thing I'd ask is if you'd viewed how this render in Markdown by building them locally and giving them a quick look. https://github.com/rapidsai/cuml/tree/branch-0.7/docs

danielhanchen commented:
Np @dantegd ! Also Thanks.
@robocopnixon I will do this, and post screenshots to verify. (I still don't have access to a CUDA machine as of yet, so I will strip the code away, and just leave the doc strings)

dantegd commented:
@cjnolet assigned you as reviewer since I know you wanted to do one final check

robocopnixon commented:
Hey @danielhanchen if you don't have a cuda machine, I think we can just merge this PR and if the docs need work someone with a cuda machine can tackle it. The package needs to be able to build to get the documents to render I believe.

danielhanchen commented:
@robocopnixon Oh. Nahh I'll still do my checks :) I'll try first. If it doesn't work, I'll get back to you ASAP.

danielhanchen commented:
@robocopnixon I noticed multiple problems with the docs: (will fix now)
1. Section headers that are not "normal" ie Applications will not work.
2. The index.rst has multiple issues that need resolving (mainly naming)
3. Will add UMAP as well.
4. Other minor issues.

Good idea on checking. I managed to make the docs compile by commenting out all code and replacing function code with just "pass".

dantegd commented:
@danielhanchen could you add a line to the changelog for the PR? Thanks!

danielhanchen commented:
By the way @dantegd , do you how I can add new files to this PR without making a new PR?

dantegd commented:
@danielhanchen Just push to the same branch in your for from which you created the PR

danielhanchen commented:
DBSCAN conflict is indentation + spacing issues. (No code changes).
However it's best if someone verify my claim.
PCA conflict is a bit confusing, since I don't see any conflicts...

danielhanchen commented:
@dantegd Thanks! Lol didn't know that.
Anyways I updated changelog, and tried to fix the conflicts.
Also fixed a tabbing/spacing issue in DBSCAN which was not compiling

danielhanchen commented:
By the way, although I am added as a member, is it still possible for full checks to be made on PRs? Cause now it only checks for Style, Changelogs

danielhanchen commented:
> By the way, although I am added as a member, is it still possible for full checks to be made on PRs? Cause now it only checks for Style, Changelogs

Don't worry now. Checks are working now

robocopnixon commented:
@dantegd can you merge this when you have a chance?
